### DISCUSSION TITLE
runner configuration without prompt

### DISCUSSION THREAD

**Discussion by moutonjeremy** [community/community#26893] [Actions]
<p>Hello,</p>
<p>When we configure a runner, he asks for the name of the runner and the workdir forlder.</p>
<p>Is it possible to configure a runner automaticly ?</p>
<p>Example : </p>
<pre><code>./config.sh --url https://github.com/toto/toto --token ALKSADLKASD:LAKSJDLSKAJLKJDSA --name linux --workdir /opt
</code></pre>
<p>Thanks</p>

**Comment by BrightRan** [comment, marked as answer]
<p>You can try the following command lines, when you install an self-hosted runner, enter the correct values of the parameters into the command lines.</p>
<p>And then just execute the command lines once, It will automatically complete to download package, extract package, and install runner according to the given parameters.</p>
<p><strong>On Windows:</strong></p>
<pre><code>Invoke-WebRequest -Uri https://githubassets.azureedge.net/runners/2.161.0/actions-runner-win-x64-2.161.0.zip -OutFile actions-runner-win-x64-2.161.0.zip; Add-Type -AssemblyName System.IO.Compression.FileSystem ; [System.IO.Compression.ZipFile]::ExtractToDirectory("$PWD/actions-runner-win-x64-2.161.0.zip", "$PWD"); ./config.cmd --url &lt;repo_url&gt; --token &lt;PAT&gt; --agent &lt;runner_name&gt; --work &lt;work_dir&gt; --runasservice
</code></pre>
<p><strong>On Linux:</strong></p>
<pre><code>curl -O https://githubassets.azureedge.net/runners/2.161.0/actions-runner-linux-x64-2.161.0.tar.gz; tar xzf ./actions-runner-linux-x64-2.161.0.tar.gz; ./config.sh --url &lt;repo_url&gt; --token &lt;PAT&gt; --agent &lt;runner_name&gt; --work &lt;work_dir&gt; --runasservice
</code></pre>
<p><strong>On macOS:</strong></p>
<pre><code>curl -O https://githubassets.azureedge.net/runners/2.161.0/actions-runner-osx-x64-2.161.0.tar.gz; tar xzf ./actions-runner-osx-x64-2.161.0.tar.gz; ./config.sh --url &lt;repo_url&gt; --token &lt;PAT&gt; --agent &lt;runner_name&gt; --work &lt;work_dir&gt; --runasservice
</code></pre>

**Reply by rafasha123** [reply]
https://githubassets.azureedge.net/runners/2.161.0/actions-runner-osx-x64-2.161.0.tar.gz

**Comment by moutonjeremy** [comment]
<p>Thanks, it’s very helpful</p>

**Comment by mkrakowitzer** [comment]
<p>–agent seems to have been replaced --name</p>

**Reply by my-wrm** [reply]
Yes

**Comment by woter1832** [comment]
<p>If you already have <code>config.sh</code> it takes the argument <code>--help</code> which outputs:</p>
<pre><code class="lang-auto">[root@myrunner github-runner]$ ./config.sh --help

Commands:
 ./config.sh         Configures the runner
 ./config.sh remove  Unconfigures the runner
 ./run.sh            Runs the runner interactively. Does not require any options.

Options:
 --help     Prints the help for each command
 --version  Prints the runner version
 --commit   Prints the runner commit

Config Options:
 --unattended           Disable interactive prompts for missing arguments. Defaults will be used for missing options
 --url string           Repository to add the runner to. Required if unattended
 --token string         Registration token. Required if unattended
 --name string          Name of the runner to configure (default myrunner)
 --runnergroup string   Name of the runner group to add this runner to (defaults to the default runner group)
 --labels string        Extra labels in addition to the default: 'self-hosted,Linux,X64'
 --work string          Relative runner work directory (default _work)
 --replace              Replace any existing runner with the same name (default false)

Examples:
 Configure a runner non-interactively:
  ./config.sh --unattended --url &lt;url&gt; --token &lt;token&gt;
 
Configure a runner non-interactively, replacing any existing runner with the same name:
  ./config.sh --unattended --url &lt;url&gt; --token &lt;token&gt; --replace [--name &lt;name&gt;]

 Configure a runner non-interactively with three extra labels:
  ./config.sh --unattended --url &lt;url&gt; --token &lt;token&gt; --labels L1,L2,L3
</code></pre>
<p>Version: <code>2.273.2</code></p>
<p>HTH</p>

**Reply by npalm** [reply]
Is there any docs that mention the requirements for the name, e.g. max length, allowed chars?
